src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/ThreadPoolTaskScheduler.cs

```diff
@@ -47,6 +47,7 @@ protected internal override void QueueTask(Task task)
                 // Run LongRunning tasks on their own dedicated thread.
 #pragma warning disable CA1416 // TODO: https://github.com/dotnet/runtime/issues/44922
                 new Thread(s_longRunningThreadWork) { IsBackground = true }.UnsafeStart(task);
+                // Do not name thread; user code may expect to do this
```

Comment:
It is fine to rename the thread. If user code sets the name, it will just get the new name.

BTW: We have a special code to deal with this on regular TP threads: The user code give the thread pool thread a new name, but we rename it back to regular TP thread name once the control flow returns back to the threadpool.

Reply:
> It is fine to rename the thread.

https://github.com/dotnet/runtime/blob/5078dd61a81c88dbdd131a77b8148b4dee3803fa/src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs#L323-L326

We could of course change that if we wanted.

Reply:
I can remove that limitation if you don't know why it exists. I wasn't sure whether debuggers/profilers would be surprised if the name changes again, but the fact that the thread pool renames itself them back suggests its ok.

Reply:
Yes, I think we should remove this limitation.